pkg/nitro_enclaves_cpu_plugin/device_plugin.go

```diff
@@ -0,0 +1,369 @@
+// Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nitro_enclaves_cpu_plugin
+
+import (
+	"errors"
+	"fmt"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s-ne-device-plugin/pkg/config"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+	deviceName                     = "nitro_enclaves_cpus"
+	devicePluginServerReadyTimeout = 10
+	deviceOfflineCPUsPath          = "/sys/devices/system/cpu/offline"
+)
+
+var cpuIdCounter = 0
+
+func (necdp *NitroEnclavesCPUDevicePlugin) socketPath() string {
+	return pluginapi.DevicePluginPath + deviceName + ".sock"
+}
+
+func (necdp *NitroEnclavesCPUDevicePlugin) resourceName() string {
+	return "aws.ec2.nitro/" + deviceName
+}
+
+type IBasicDevicePlugin interface {
+	Start() error
+	Stop()
+	socketPath() string
+	resourceName() string
+}
+
+// NitroEnclavesCPUDevicePlugin implements the Kubernetes device plugin API
+type NitroEnclavesCPUDevicePlugin struct {
+	devices []*pluginapi.Device
+
+	stop chan interface{}
+
+	server *grpc.Server
+	pluginapi.DevicePluginServer
+	IBasicDevicePlugin
+}
+
+// FileReader interface defines the behavior for reading CPU information
+type fileReader interface {
+	read() (string, error)
+}
+
+// SystemFileReader implements FileReader for the actual system
+type systemFileReader struct {
+	path string
+}
+
+func newSystemFileReader(path string) *systemFileReader {
+	return &systemFileReader{path: path}
+}
+
+func (s *systemFileReader) read() (string, error) {
+	data, err := os.ReadFile(s.path)
+	if err != nil {
+		return "", fmt.Errorf("error reading data from %s: %v", s.path, err)
+	}
+	return string(data), nil
+}
+
+// determineAdvisableCPUs reads the number of offline cpus from /sys/devices/system/cpu/offline
+func determineAdvisableCPUs(fileReader fileReader) (int, error) {
+
+	data, err := fileReader.read()
+	if err != nil {
+		return 0, fmt.Errorf("error reading offline CPU file: %v", err)
+	}
+
+	// Handle empty/unknown case
+	content := strings.TrimSpace(string(data))
+	if content == "" || content == "-1" {
```

Comment:
is `-1` a valid content for that file?

Reply:
was supposed to be tested in later section in case that invalid CPU tuple was written to file such as `5-6,-6` however has been covered in is valid number check